src/object.d

```diff
@@ -3321,23 +3310,22 @@ int __cmp(T1, T2)(T1[] s1, T2[] s2)
         import core.internal.string : dstrcmp;
         return (() @trusted => dstrcmp(cast(char[])s1, cast(char[])s2))();
     }
-    // Everything else
+    // Everything else: class, interface, struct, other built-ins
     else
     {
         immutable len = s1.length <= s2.length ? s1.length : s2.length;
 
         foreach (const u; 0 .. len)
         {
-            // structs
-            static if (__traits(compiles, at(s1, u).opCmp(at(s2, u))))
+            static if (__traits(compiles, __cmp(at(s1, u), at(s2, u))))
```

Comment:
I wanted to make sure `__cmp` gets first crack so we don't call `a.opCmp(b)` where a is a class type and its value is null.